pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1371,7 +1381,12 @@ def test_merge_readonly(self):
             np.arange(20).reshape((5, 4)) + 1, columns=["a", "b", "x", "y"]
         )
 
-        data1._mgr.blocks[0].values.flags.writeable = False
+        if using_array_manager:
+            for arr in data1._mgr.arrays:
+                arr.flags.writeable = False
+        else:
+            data1._mgr.blocks[0].values.flags.writeable = False
```

Comment:
could implement BlockManager.arrays as a property yielding the values for compat

Reply:
Yeah, but that would not help for this specific case (as those arrays would not be the actual arrays stored in the BlockManager, and setting the flag wouldn't have an impact on the actual values).

Reply:
> as those arrays would not be the actual arrays stored in the BlockManager

```
@property
def arrays(self):
   for blk in self.blocks:
       yield blk.values
```

Reply:
Ah, like that. I wouldn't find it great API design, as the same attribute/property would return something quite different (at least what you can do with it is different), but I am fine adding it for testing convenience

Reply:
Added